pandas/io/parsers.py

```diff
@@ -2223,13 +2223,14 @@ def _clean_index_names(columns, index_col):
 def _get_empty_meta(columns, index_col, index_names):
     columns = list(columns)
 
-    if index_col is not None:
-        index = MultiIndex.from_arrays([[]] * len(index_col),
-                                       names=index_names)
-        for n in index_col:
-            columns.pop(n)
-    else:
+    if index_col is None or index_col is False:
         index = Index([])
+    else:
+        index_col = list(index_col)
+        index = MultiIndex.from_arrays([[]] * len(index_col), names=index_names)
+        index_col.sort()
```

Comment:
In this PR we already have tests that specify a multiindex with either a list of strings or a list of ints (see `test_empty_with_multiindex` and `test_empty_with_reversed_multiindex`).

I haven't completely stepped through the code but afaics, function `_clean_index_names` is called prior to `get_empty_meta` and converts `index_col` as a list of strings into a list of ints.